services/web/server/src/simcore_service_webserver/invitations_core.py

```diff
@@ -0,0 +1,118 @@
+"""
+    Plugin to interact with the invitations service
+"""
+
+import logging
+from contextlib import contextmanager
+
+import sqlalchemy as sa
+from aiohttp import ClientError, ClientResponseError, web
+from pydantic import ValidationError
+from pydantic.errors import PydanticErrorMixin
+from simcore_postgres_database.models.users import users
+
+from .db import get_database_engine
+from .invitations_client import (
+    InvitationContent,
+    InvitationsServiceApi,
+    get_invitations_service_api,
+)
+
+logger = logging.getLogger(__name__)
+
+
+#
+# DATABASE
+#
+
+
+async def _is_user_registered(app: web.Application, email: str) -> bool:
+    pg_engine = get_database_engine(app=app)
+
+    async with pg_engine.acquire() as conn:
+        user_id = await conn.scalar(
+            sa.select([users.c.id]).where(users.c.email == email)
+        )
+        return user_id is not None
+
+
+#
+# API plugin ERRORS
+#
+
+
+class InvitationsErrors(PydanticErrorMixin, ValueError):
+    ...
+
+
+class InvalidInvitation(InvitationsErrors):
+    msg_template = "Invalid invitation: {reason}"
+
+
+class InvitationsServiceUnavailable(InvitationsErrors):
+    msg_template = (
+        "Unable to process your invitation since the invitations service is currently unavailable. "
+        "Please try again later."
+    )
+
+
+@contextmanager
+def _handle_exceptions_as_invitations_errors():
+    try:
+
+        yield  # API function calls happen
+
+    except ClientResponseError as err:
+        # check possible errors
+        if err.status == web.HTTPUnprocessableEntity.status_code:
+            raise InvalidInvitation(reason=err.message) from err
+
+        # some validation or other error?
+        raise InvitationsServiceUnavailable() from err
+
+    except ValidationError as err:
+        raise InvitationsServiceUnavailable() from err
+
+    except ClientError as err:
+        raise InvitationsServiceUnavailable() from err
+
+    except InvitationsErrors:
+        # bypass
+        raise
```

Comment:
Let's see if I read it carefully this should raise `InvitationsErrors` otherwise these will get caught by the block below. Is this the intent? If yes, maybe have a different comment, this one is not that helpful

Reply:
yes, that is the intent: it **bypasses** the more general exception handler.